snapmaker/src/gcode/M2001.cpp

```diff
@@ -0,0 +1,74 @@
+/*
+ * Snapmaker2-Controller Firmware
+ * Copyright (C) 2019-2020 Snapmaker [https://github.com/Snapmaker]
+ *
+ * This file is part of Snapmaker2-Controller
+ * (see https://github.com/Snapmaker/Snapmaker2-Controller)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "snapmaker.h"
+#include "src/gcode/gcode.h"
+
+/*
+ * Power domain control:
+ *   S: just for screen
+ *   T: for all executors and all linear modules
+ *   B: for bed and addon
+ * 
+ * Call with:
+ *   M2001 S{0} T{0} B{0}
+ */
+void GcodeSuite::M2001() {
+  bool reboot_req = false, val;
+  if (parser.seen('S')) {
```

Comment:
here you can confirm the system status by below condition
(system.GetCurrentStatus() != SYSTAT_IDLE)

Reply:
Thank you, agreed.